adafruit_emc2101/__init__.py

```diff
@@ -232,30 +266,59 @@ def initialize(self):
         self._spin_tach_limit = False
         self._calculate_full_speed()
 
+    def check_status(self):
+        """Read the status register and check for a fault indicated."""
+        if self._status & emc2101_regs.STATUS_ALERT:
+            raise ValueError("Status alert")
+
+    @property
+    def part_info(self):
+        """The part information: manufacturer, part id and revision. Normally (0x5d, 0x16, 0x1)"""
+        return (self._mfg_id, self._part_id, self._part_rev)
+
+    @property
+    def devconfig(self):
+        """Read device config register."""
+        return self._config
+
+    @property
+    def devstatus(self):
+        """Read device status register."""
+        return self._status
+
     @property
     def internal_temperature(self):
         """The temperature as measured by the EMC2101's internal 8-bit temperature sensor"""
+        self.check_status()
         return self._int_temp
 
     @property
     def external_temperature(self):
         """The temperature measured using the external diode"""
 
+        self.check_status()
         temp_lsb = self._ext_temp_lsb
         temp_msb = self._ext_temp_msb
         full_tmp = (temp_msb << 8) | temp_lsb
         full_tmp >>= 5
-        full_tmp *= 0.125
+        if full_tmp == emc2101_regs.TEMP_FAULT_OPENCIRCUIT:
+            raise ValueError("Open circuit")
+        if full_tmp == emc2101_regs.TEMP_FAULT_SHORT:
+            raise ValueError("Short circuit")
 
-        return full_tmp
+        full_tmp *= 0.125
+        return float(full_tmp)
 
     @property
     def fan_speed(self):
         """The current speed in Revolutions per Minute (RPM)"""
+        self.check_status()
```

Comment:
What kind of fault is is this checking for? 

I'm noticing that with my test setup this version of the driver from the PR ends up raising an exception from this whereas the currently released version seems to allow normal operation.

I'm testing with Feather RP2040 and a small 5V fan like this one: https://www.amazon.com/gp/product/B092YXQMX5?th=1 

It doesn't have a tachometer so I don't have anything hooked up to that pin on the controller. Could that be the cause of this fault exception?